src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2776,7 +2793,7 @@ private MemberDeclarationSyntax ParseMemberDeclarationCore(SyntaxKind parentKind
 
             try
             {
-                var attributes = this.ParseAttributeDeclarations();
+                var attributes = this.ParseAttributeDeclarations(inExpressionContext: false);
```

Comment:
all the places taht pass in false should have tests taht show we still think tehy are an attribute, even if the user types `.` after them.